docs/js-core-3/week-3/lesson.mdx

```diff
@@ -287,319 +285,168 @@ function six() {
-myAddressBook.store("maggie", "maggie@simpsons.com");
+```js
+let student = ["Sara", "Lee", 30, "London", "Javascript"];
 
-myAddressBook.lookup("bart");
-// 'bart@simpsons.com'
+let [firstName, lastName, age, city, subject] = student;
 
-myAddressBook.lookup("homer");
-// 'address not found'
+console.log(
+  `My name is ${firstName} ${lastName}. I am ${age} years old. I live in ${city} and I am studying ${subject}.`
+);
 ```
 
-Notice that when you come to using a class instance, you are only interacting with it through its methods, e.g. `store` and `lookup` for `AddressBook`. Similar to functions, classes provide a way of creating abstractions over lower-level implementations, which can help us better organise and maintain our code.
+With array destructuring you were able to reduce this code down from 7 lines of code to just 3 lines of code. This means your code is shorter and there is less chance for you to have a typo or a bug in your code.
 
-## 3. `this` keyword
+#### Exercise (2)
 
-One aspect of JavaScript that often leads to confusion is the keyword `this`. Its value depends on where it is used and how that function is called.
+Rewrite the code below to use array destructuring instead of assigning each value to a variable.
 
-### `this` in a constructor
+<!-- These imports are used to enable this feature. Do not delete. -->
+<!-- https://v2.docusaurus.io/docs/markdown-features#multi-language-support-code-blocks -->
 
-We saw previously that a `constructor` in a `class` is a special method which is used to instantiate on a new object. When used inside a `constructor`, `this` refers to the new object being created.
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+<Tabs
+  defaultValue="exercise"
+  values={[
+    { label: 'Exercise', value: 'exercise', },
+    { label: 'Solution', value: 'solution', },
+  ]
+}>
+<TabItem value="exercise">
 
 ```js
-class Person {
-  constructor() {
-    console.log(this);
-    this.name = "Mike";
-  }
-}
+let sentence = ["Hello", "Code", "Your", "Future"];
 
-const person = new Person();
-```
+let firstWord = item[0];
+let secondWord = item[1];
+let thirdWord = item[2];
+let fourthWord = item[3];
 
-### `this` in a method
+console.log(
+  `FirstWord: ${firstWord}, SecondWord: ${secondWord}, ThirdWord: ${thirdWord}, FourthWord: ${fourthWord},`
+);
+```
 
-A method is a function that belongs to an object. That object can be either be created as an object literal or using a class.
+</TabItem>
+<TabItem value="solution">
 
 ```js
-class Person {
-  constructor() {
-    this.name = "Mike";
-  }
+let sentence = ["Hello", "Code", "Your", "Future"];
 
-  sayHi() {
-    console.log(`Hi, ${this.name}`);
-  }
-}
+let [firstWord, secondWord, thirdWord, fourthWord] = sentence;
 
-const person = new Person();
-person.sayHi();
+console.log(
+  `FirstWord: ${firstWord}, SecondWord: ${secondWord}, ThirdWord: ${thirdWord}, FourthWord: ${fourthWord},`
+);
```

Comment:
It might be worth adding another example which shows that values of any type can be destructured, as this will be important when it comes to `useState`. I think we only show strings and numbers here at the moment.

Perhaps something like:

```js
let citiesByPopulation = [
  { name: "London", population: 8000000 },
  { name: "Birmingham", population: 1000000 },
  { name: "Glasgow", population: 600000 },
  { name: "Manchester", population: 500000 },
];

console.log(`${london.name} population is ${london.population}`);
console.log(`${birmingham.name} population is ${birmingam.population}`);
console.log(`${glasgow.name} population is ${glasgow.population}`);
console.log(`${manchester.name} population is ${manchester.population}`);
```

?